### PR TITLE
neuralcoref: init at 4.0

### PR DESCRIPTION
**Pull request by ConradMearns**
###### Motivation for this change
Add the neuralcoref module to python3Packages

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I have not yet "Tested execution of all binary files" by running the module in Python yet. Been overloaded at work :upside_down_face: 

Let me know if there's anything I can improve or should change, thanks!

### CODE REVIEW

**Comment by jonringer** [comment]
when updating a branch, please use rebase. E.g. `git pull -r .. ...` . This helps keep the git history cleaner.

File: pkgs/development/python-modules/neuralcoref/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  doCheck = false;
+  # checkPhase = ''
+  #   ${python.interpreter} -m pytest spacy/tests --vectors --models --slow
+  # '';
+
+  meta = with lib; {
+    description = "Neuralcoref: Coreference Resolution in spaCy with Neural Networks";
+    homepage = "https://github.com/huggingface/neuralcoref";
+    license = licenses.mit;
+    maintainers = with maintainers; [ conradmearns ];
```
Comment (jonringer) [inline pkgs/development/python-modules/neuralcoref/default.nix:39]: Please create a separate commit that adds you to maintainers.nix. Look at https://github.com/NixOS/nixpkgs/commit/261912e07124bf8e72cca2049c2a66acc0288db7 as an example

```
maintainers: add <name>
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7426,6 +7426,8 @@ in {
 
   rxv     = callPackage ../development/python-modules/rxv     { };
 
+  neuralcoref = callPackage ../development/python-modules/neuralcoref {};
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:7429]: please attempt to sort this

File: pkgs/development/python-modules/neuralcoref/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  checkInputs = [
+    pytest
+  ];
+
+  doCheck = false;
+  # checkPhase = ''
+  #   ${python.interpreter} -m pytest spacy/tests --vectors --models --slow
+  # '';
```
Comment (jonringer) [inline pkgs/development/python-modules/neuralcoref/default.nix:33]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

**Comment by ConradMearns** [comment]
@jonringer, sorry for the mess and late reply, I only partially know what I am doing.

It may take me a while to be ready to create tests for this, in the meantime I will add the maintainer entry via a separate package PR

Should I and revert or change the git history to rebase? I'm assuming I would need to do this anyway to fix the mess with the maintainer entry, but I don't know how to do this yet.

**Comment by abathur** [comment]
@ConradMearns Are you still interested in pursuing this? I see you subsequently submitted and maintain the obsidian package, so maybe the places where this gummed up are more tractable now?

**Comment by abathur** [comment]
Tentatively closing. Happy to reopen if you want to pursue it.

**Pull request closed without merging**